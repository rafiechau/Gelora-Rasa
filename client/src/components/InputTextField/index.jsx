import { FormControl } from '@mui/material';
import PropTypes from 'prop-types';
import classes from './style.module.scss';

const InputTextField = ({ children, input, disabled = false, register, errors, accept = undefined }) => (
  <FormControl>
    <label htmlFor={input?.name} className={classes.labelInput}>
      {input?.label}
      <input
        className={`${classes.input} ${errors[input?.name] && classes.inputError}`}
        type={input?.type}
        {...register(input?.name, {
          required: input?.required,
          pattern: { value: input?.pattern, message: input?.messagePatern },
          minLength: {
            value: input?.minLength,
            message: input?.messageMin,
          },
        })}
        id={input?.name}
        placeholder={input?.placeholder}
        defaultValue={input?.value}
        accept={accept && accept}
        disabled={disabled}
      />
      {errors[input?.name] && <span className={classes.inputLabelError}>{errors[input?.name].message}</span>}
      {children}
    </label>
  </FormControl>
);
InputTextField.propTypes = {
  children: PropTypes.element,
  input: PropTypes.shape({
    name: PropTypes.string,
    placeholder: PropTypes.string,
    label: PropTypes.string,
    required: PropTypes.string,
    pattern: PropTypes.shape(RegExp),
    type: PropTypes.string,
    value: PropTypes.string,
    messagePatern: PropTypes.string,
    messageMin: PropTypes.string,
    minLength: PropTypes.number,
  }),
  register: PropTypes.func,
  errors: PropTypes.object,
  accept: PropTypes.string,
  disabled: PropTypes.bool,
};

export default InputTextField;
