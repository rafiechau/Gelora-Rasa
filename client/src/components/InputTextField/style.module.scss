.labelInput{
    color: white; 
    font-size: 1rem; 
    margin-bottom: 0.5rem; 
    .input{
        width: 100%;
        padding: 0.75rem;
        margin-bottom: 1rem; 
        border-radius: 5px;
        border: 1px solid lighten($navy, 20%);
        background-color: #333;
        color: #ddd;
        box-sizing: border-box;
        &:focus {
            outline: none;
        }
    }
    .inputError {
        border: 1px solid #ed0226;
    }
}
.inputLabel {
    @include directionX;
    width: 100%;
    padding: 0.5rem 0 0.5rem 0;
    font-weight: 400;
    font-size: small;
}
.inputLabelError {
    color: #ed0226;
    font-size: small;
}