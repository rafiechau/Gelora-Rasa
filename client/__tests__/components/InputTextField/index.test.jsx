import React from 'react';
import { render } from '@testing-library/react';
import InputTextField from '@components/InputTextField';

// Contoh properti untuk InputTextField
const mockInputProps = {
  name: 'testName',
  placeholder: 'Enter text',
  label: 'Test Label',
  required: 'This field is required',
  pattern: /some-regex/,
  type: 'text',
  value: '',
  messagePattern: 'Invalid pattern',
  messageMin: 'Too short',
  minLength: 3,
};

const mockRegister = () => {};

const mockErrors = {};

describe('InputTextField', () => {
  it('renders correctly and matches snapshot', () => {
    const { asFragment } = render(
      <InputTextField
        input={mockInputProps}
        register={mockRegister}
        errors={mockErrors}
        data-testid="input-text-field"
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
