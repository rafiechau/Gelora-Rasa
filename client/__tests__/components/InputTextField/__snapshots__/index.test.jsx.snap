// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputTextField renders correctly and matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
    data-testid="input-text-field"
  >
    <label
      class="labelInput"
      for="testName"
    >
      Test Label
      <input
        class="input undefined"
        id="testName"
        placeholder="Enter text"
        type="text"
        value=""
      />
    </label>
  </div>
</DocumentFragment>
`;
